src/sys/options.cxx

```diff
@@ -45,8 +45,36 @@ Options::Options(const char* value) {
 }
 
 Options::Options(std::initializer_list<std::pair<std::string, Options>> values) {
+  // Yes, this looks bad, but bear with me...  The _only_ way to
+  // construct a nested initializer_list is inside-out, from the
+  // bottom of the tree structure. Unfortunately, this is very much
+  // not how you would want to construct the tree of options, as we
+  // don't have the parent section's name as we construct each
+  // child. Therefore, when we _do_ construct the parent, we'll need
+  // to recursively step down the tree, prepending the parent's name
+  // to each child. Rather than have a private member to do that, we
+  // use a lambda. And to make that lambda recursive, we need to have
+  // a nested lambda.
+  auto append_section_name = [](auto& children, const std::string& section_name) {
+    auto append_impl = [](auto& children, const std::string& section_name, auto& append_ref) mutable -> void {
+      for (auto& child : children) {
+        child.second.full_name = fmt::format("{}:{}", section_name, child.second.full_name);
+        if (child.second.is_section) {
+          append_ref(child.second.children, section_name, append_ref);
+        }
+      }
+    };
+    append_impl(children, section_name, append_impl);
+  };
+
```

Comment:
Is it worth instead making full_name a function, which iterates through its parents? Of course that raises the problem of setting pointers to parents... Another possibility is to not store the full name or parent, but that then puts it on wherever the full name is needed to work it out.